Tasks/S3Download/TaskOperations.ts

```diff
@@ -69,7 +69,9 @@ export class TaskOperations {
                 }
                 if (this.taskParameters.keyManagement === customerManagedKeyValue) {
                     params.SSECustomerAlgorithm = aes256AlgorithmValue
-                    params.SSECustomerKey = this.taskParameters.customerKey
+                    if (this.taskParameters.customerKey !== Buffer.from([])) {
```

Comment:
Is there another way to verify? For example, if we can't compare against something like length, could we create a static readonly instance of the initialized value used in TaskParameters.ts (line 38) and compare against that.

Reply:
We can check length (not always but in this case we can)